rajawali/src/main/java/org/rajawali3d/renderer/RajawaliPipRenderer.java

```diff
@@ -0,0 +1,192 @@
+ * <code>SubRenderer</code>s through <code>setMainRenderer</code> (for the main content)
+ * <code>setMinirenderer</code> (for the content in the mini view).
+ * The <code>pipScale</code> and <code>pipMargin</code> parameters control how the mini scene
+ * is placed.
+ * </p>
+ */
+public class RajawaliPipRenderer extends RajawaliRenderer {
+    private static final String TAG = "RajawaliPipRenderer";
+
+    private RenderTarget mMainRenderTarget;
+    private RenderTarget mMiniRenderTarget;
+
+    private WorkaroundScreenQuad mMiniQuad;
+    private WorkaroundScreenQuad mMainQuad;
+    private Material mMiniQuadMaterial;
+    private Material mMainQuadMaterial;
+
+    private RajawaliScene mCompositeScene;
+
+    private SubRenderer mMiniRenderer;
+    private SubRenderer mMainRenderer;
+
+    // These variables control where the minimap is placed. Note they are specified in standardized
+    // OpenGL coordinates 0 to 1
+    private final float pipScale;
+    private final float pipMarginX;
+    private final float pipMarginY;
+
+    private float miniXmin, miniXmax, miniYmin, miniYmax;
+
+    /**
+     * @param pipScale   Size of the mini view from 0 to 1, i.e.: ratio of full screen to mini
+     * @param pipMarginX Margin space from the mini view to the edge of the main view in pixels
+     * @param pipMarginY Margin space from the mini view to the edge of the main view in pixels
+     */
+    public RajawaliPipRenderer(Context context, float pipScale, float pipMarginX, float pipMarginY) {
+        super(context);
+        this.pipScale = pipScale;
+        this.pipMarginX = pipMarginX;
+        this.pipMarginY = pipMarginY;
+    }
+
+    public void setMiniRenderer(SubRenderer mMiniRenderer) {
+        this.mMiniRenderer = mMiniRenderer;
+    }
+
+    public void setMainRenderer(SubRenderer mMainRenderer) {
+        this.mMainRenderer = mMainRenderer;
+    }
+
+    @Override
+    public void initScene() {
+        mMainQuadMaterial = new Material();
+        mMainQuadMaterial.setColorInfluence(0);
+
+        mMiniQuadMaterial = new Material();
+        mMiniQuadMaterial.setColorInfluence(0);
+
+        mMainQuad = new WorkaroundScreenQuad();
+        mMainQuad.setMaterial(mMainQuadMaterial);
+        mMainQuad.setTransparent(true);
+
+        // Set-up viewport dimensions of mini quad for touch event processing
+        setupMiniTouchLimits();
+
+        mMiniQuad = new WorkaroundScreenQuad();
+        // Set the size of the mini view using a scale factor (pipScale times the main view)
+        mMiniQuad.setScale(pipScale);
+        // Position the mini view in the top right corner
+        // For X and Y, the position is:
+        //   50% screen shift to the right/top minus half the size of the minimap to bring it back
+        //   left/bottom into full view plus a little bit more left/bottom to leave margin
+        mMiniQuad.setX(.5 - pipScale / 2 - pipMarginX / (2 * mDefaultViewportWidth));
```

Comment:
I would prefer mixing floats/doubles/integers as it can cause logic issues and confuse intent. Can you please update the .5 to .5d and specify the 2 instances as 2d unless of course this needs to all be float.